src/ui/preview/PlaygroundHttpProxy.ts

```diff
@@ -0,0 +1,45 @@
+import {
+	PHPRequest,
+	PHPResponse,
+	PlaygroundClient,
+} from '@wp-playground/client';
+
+export class PlaygroundHttpProxy {
+	constructor( private readonly client: PlaygroundClient ) {}
+
+	async request( request: PHPRequest ): Promise< PHPResponse > {
+		const response = await this.client.request( request );
+
+		if ( response.httpStatusCode < 200 || response.httpStatusCode >= 300 ) {
+			logFailedRequest( { request, response } );
+		} else if ( process.env.LOG_REQUESTS === 'true' ) {
+			logRequest( { request, response } );
+		}
+
+		return response;
+	}
+}
+
+function logRequest( args: { request: PHPRequest; response: PHPResponse } ) {
+	const { request, response } = args;
+	const url = request.url;
+	const message = `[Request] ${ url }
+[Request body] ${ request.body }
+[Response] (${ response.httpStatusCode })`;
+	console.log( message, response.json );
```

Comment:
```suggestion
	console.log( {
		type: 'API Request/Response',
		request: {
			url,
			body:
				typeof request.body === 'string'
					? ( () => {
							try {
								return JSON.parse( request.body );
							} catch {
								return request.body;
							}
					  } )()
					: request.body,
		},
		response: { status: response.httpStatusCode, body: response.json },
	} );
```

Reply:
The suggested change gets us this:

<img width="909" alt="Screenshot 2024-12-19 at 16 48 46" src="https://github.com/user-attachments/assets/9fe8ed34-a5b1-4a50-abe8-9ec3e3e16517" />